sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Grpc/SettlementService.cs

```diff
@@ -127,6 +127,105 @@ public override async Task<Empty> Deadletter(DeadletterRequest request, ServerCa
             throw new RpcException (new Status(StatusCode.FailedPrecondition, $"LockToken {request.Locktoken} not found."));
         }
 
+        public override async Task<Empty> RenewMessageLock(RenewMessageLockRequest request, ServerCallContext context)
+        {
+            try
+            {
+                if (_provider.ActionsCache.TryGetValue(request.Locktoken, out var tuple))
+                {
+                    await tuple.Actions.RenewMessageLockAsync(
+                        tuple.Message,
+                        context.CancellationToken).ConfigureAwait(false);
+                    return new Empty();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new RpcException(new Status(StatusCode.Unknown, ex.ToString()));
+            }
+
+            throw new RpcException(new Status(StatusCode.FailedPrecondition, $"LockToken {request.Locktoken} not found."));
+        }
+
+        public override async Task<GetSessionStateResponse> GetSessionState(GetSessionStateRequest request, ServerCallContext context)
+        {
+            try
+            {
+                if (_provider.SessionActionsCache.TryGetValue(request.SessionId, out var actions))
+                {
+                    var sessionState = await actions.GetSessionStateAsync(context.CancellationToken).ConfigureAwait(false);
+                    return new GetSessionStateResponse
+                    {
+                        SessionState = ByteString.CopyFrom(sessionState)
+                    };
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new RpcException(new Status(StatusCode.Unknown, ex.ToString()));
+            }
+
+            throw new RpcException(new Status(StatusCode.FailedPrecondition, $"SessionId {request.SessionId} not found."));
+        }
+
+        public override async Task<Empty> SetSessionState(SetSessionStateRequest request, ServerCallContext context)
+        {
+            try
+            {
+                if (_provider.SessionActionsCache.TryGetValue(request.SessionId, out var actions))
+                {
+                    await actions.SetSessionStateAsync(BinaryData.FromString(request.SessionState.ToStringUtf8()),
```

Comment:
I don't think we should be converting to a string as the data is binary.

Reply:
Changed!